cl/lib/elasticsearch_utils.py

```diff
@@ -433,21 +437,66 @@ def build_text_filter(field: str, value: str) -> List:
     return []
 
 
-def build_sort_results(cd: CleanData) -> Dict:
+def toggle_sort_order(
+    order_by: str | None, toggle_sorting: bool
+) -> str | None:
+    """Toggle the sorting order of the given "order_by" string from ascending
+    to descending, or vice versa. This is used for changing the sort direction
+    in queries, useful to perform backward pagination for the V4 Search API.
+
+    :param order_by: A string specifying the fields and sort directions,
+    separated by commas, e.g., "score asc, score desc".
+    :param toggle_sorting: A boolean flag that indicates whether the sorting
+    order should be toggled. If False, the original "order_by" is returned.
+    :return: A modified "order_by" string with toggled sort directions.
+    """
+
+    if toggle_sorting and order_by:
+        sort_components = order_by.split(",")
+        toggle_sort_components = []
+        for component in sort_components:
+            component = component.strip()
+            if "desc" in component:
+                toggle_sort_components.append(component.replace("desc", "asc"))
+            elif "asc" in component:
+                toggle_sort_components.append(component.replace("asc", "desc"))
+            else:
+                toggle_sort_components.append(component)
+
+        return ", ".join(toggle_sort_components)
+
+    return order_by
```

Comment:
We can simplify the code structure and improve readability by using an early return to reduce the size of the nested block. (we should have just one nested line after the refactor)

Reply:
Sure, updated it.